webui/react/src/pages/ExperimentList.tsx

```diff
@@ -228,6 +233,28 @@ const ExperimentList: React.FC<Props> = ({ project }) => {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [id, isLoadingSettings, settings, labelsString, pinnedString, statesString, usersString]);
 
+  const fetchHiddenSelectedExperiments = useCallback(async (): Promise<void> => {
+    try {
+      if ((settings.row?.length ?? 0) > 0) {
+        const fetchedExpIds = new Set(experiments.map((exp) => exp.id));
+        const hiddenExpIds = (settings.row ?? []).filter((id) => !fetchedExpIds.has(id));
+        const chunkedSelectedExperimentIds = _.chunk(hiddenExpIds, settings.tableLimit);
+        const selectedExpriments = await Promise.all(
+          chunkedSelectedExperimentIds.map(async (ids) => {
+            const expResponse = await getExperiments(
+              { experimentIdFilter: { incl: ids } },
+              { signal: canceler.current.signal },
+            );
+            return expResponse.experiments;
+          }),
+        );
+        setHiddenSelectedExperiments(selectedExpriments.flat());
```

Comment:
if we're using `settings.row` length as a conditional before, might as well also check `hiddenExpIds` length?

Reply:
good catch. fixed